src/DotNetWorker.Grpc/GrpcServiceCollectionExtensions.cs

```diff
@@ -50,7 +52,7 @@ public static IServiceCollection AddGrpc(this IServiceCollection services)
             services.AddSingleton<IWorkerDiagnostics, GrpcWorkerDiagnostics>();
 
             // FunctionMetadataProvider for worker driven function-indexing
-            services.AddSingleton<IFunctionMetadataProvider, DefaultFunctionMetadataProvider>();
+            services.TryAddSingleton<IFunctionMetadataProvider, DefaultFunctionMetadataProvider>();
```

Comment:
I see we are going with only a single provider. This would mean that registering of services is sensitive to the order it is done in, which may cause some obscure issues in user code if done incorrectly.

Would it be possible/useful to allow multiple providers and to aggregate their results? If so, can we open an issue to do this in a later PR? I am not super familiar with this logic, so I would not be surprised in it is not possible to support multiple providers (no way to dedupe functions maybe?).

Reply:
What we want here is either one provider or the other - there's no scenario/cases where we should aggregate the results. I agree that an issue can/should be opened so it isn't sensitive to the order of registration. 

I think when I discussed registering this with @fabiocav that if our guidance for the preview phase was followed exactly the ordering would work, but for a second preview/GA we should put in a change.

Reply:
Thanks for the explanation. Feel free to ignore this suggestion.